core/src/main/java/ai/timefold/solver/core/impl/solver/scope/SolverScope.java

```diff
@@ -62,6 +67,11 @@ public class SolverScope<Solution_> {
      */
     private final Map<Tags, List<AtomicReference<Number>>> stepScoreMap = new ConcurrentHashMap<>();
 
+    /**
+     * Used for tracking move count per move type
+     */
+    private final Map<String, AtomicLong> moveEvaluationCountPerTypeMap = new ConcurrentHashMap<>();
```

Comment:
If the map is already concurrent, do you need the `AtomicLong`?
Wouldn't locking the entire write operation be a much better guarantee of atomicity?

Reply:
> If the map is already concurrent, do you need the AtomicLong?

Considering no metric collects this information over time and the map is not being updated concurrently, we can use the Long instead.

> Wouldn't locking the entire write operation be a much better guarantee of atomicity?

Since we are using ConcurrentHashMap and atomicity is no longer necessary, I don't see a reason to do that.